src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,131 @@
+package mate.academy.dao;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String CREATE_QUERY = "INSERT INTO books (title, price) VALUES (?, ?)";
+    private static final String FIND_BY_ID_QUERY = "SELECT * FROM books WHERE id = ?";
+    private static final String FIND_ALL_QUERY = "SELECT * FROM books";
+    private static final String UPDATE_QUERY = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_BY_ID_QUERY = "DELETE FROM books WHERE id = ?";
+    private static final String URL =
+            "jdbc:mysql://localhost:3306/book_db?user=root&password=12345";
+    private final Connection connection;
+
+    public BookDaoImpl() {
+        try {
+            this.connection = DriverManager.getConnection(URL);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Could not connect to the database", e);
+        }
+    }
```

Comment:
Your solution has drawbacks. 
1. The connection should be closed (try wth resources)
2. all your operations work in a single connection
```suggestion
    public BookDaoImpl() {
        try {
            this.connection = DriverManager.getConnection(URL);
        } catch (SQLException e) {
            throw new DataProcessingException("Could not connect to the database", e);
        }
    }
```